subsys/usb/device/class/cdc_acm.c

```diff
@@ -278,6 +283,13 @@ static void cdc_acm_read_cb(uint8_t ep, int size, void *priv)
 		LOG_ERR("Ring buffer full, drop %zd bytes", size - wrote);
 	}
 
+	if(ring_buf_space_get(dev_data->rx_ringbuf) < CDC_ACM_BUFFER_SIZE)
```

Comment:
We have made API for CDC usage and during basic tests by sending data via any serial terminal everything was ok but when we made PC app to do stress test and check data consistency for this API it turns out that we lost some data. We observed that data is lost because driver do not stop data reception when the ring buffer is full.

I can make changes as you propose to remove that variable.

Reply:
ooh, you run out of 1024 bytes UART ring buffer :-)